setup.py

```diff
@@ -158,13 +159,16 @@ def run(self):
     opt_level = os.getenv("MYPYC_OPT_LEVEL", "3")
     debug_level = os.getenv("MYPYC_DEBUG_LEVEL", "1")
     force_multifile = os.getenv("MYPYC_MULTI_FILE", "") == "1"
-    ext_modules = mypycify(
-        mypyc_targets + ["--config-file=mypy_bootstrap.ini"],
-        opt_level=opt_level,
-        debug_level=debug_level,
-        # Use multi-file compilation mode on windows because without it
-        # our Appveyor builds run out of memory sometimes.
-        multi_file=sys.platform == "win32" or force_multifile,
+    ext_modules = cast(
```

Comment:
Why do we need the `cast` here?

Reply:
`setup` is imported from `setuptools`, the `ext_modules` param expects a `Sequence[setuptools.Extension]` but `mypycify` can return a list of either `setuptools.Extension` or `distutils.core.Extension`.

Reply:
I'd strongly prefer to avoid casts here, since they don't work well with mypyc. Can we use an assertion instead?

Reply:
I sure can. It's a list so I think I'd have to use a typeguard. Hopefully that's fine.

Reply:
> I'd strongly prefer to avoid casts here, since they don't work well with mypyc.

Is the setup.py compiled by mypyc?